dff/messengers/telegram/abstract.py

```diff
@@ -0,0 +1,631 @@
+                                            "filename",
+                                            "caption",
+                                            "parse_mode",
+                                            "supports_streaming",
+                                            "has_spoiler",
+                                            "thumbnail",
+                                        ],
+                                    ),
+                                ),
+                            ]
+                        else:
+                            await bot.send_video(
+                                chat_id,
+                                attachment_bytes,
+                                caption=message.text,
+                                **generate_extra_fields(
+                                    attachment,
+                                    [
+                                        "disable_notification",
+                                        "reply_markup",
+                                        "parse_mode",
+                                        "supports_streaming",
+                                        "has_spoiler",
+                                        "thumbnail",
+                                        "filename",
+                                        "message_effect_id",
+                                    ],
+                                ),
+                            )
+                            message_text_covered = True
+                if isinstance(attachment, Animation):
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        if media_group_attachments_num != 1:
+                            files += [
+                                InputMediaAnimation(
+                                    attachment_bytes,
+                                    **generate_extra_fields(
+                                        attachment, ["filename", "caption", "parse_mode", "has_spoiler", "thumbnail"]
+                                    ),
+                                ),
+                            ]
+                        else:
+                            await bot.send_animation(
+                                chat_id,
+                                attachment_bytes,
+                                caption=message.text,
+                                **generate_extra_fields(
+                                    attachment,
+                                    [
+                                        "parse_mode",
+                                        "disable_notification",
+                                        "reply_markup",
+                                        "has_spoiler",
+                                        "thumbnail",
+                                        "filename",
+                                        "message_effect_id",
```

Comment:
`message_effect_id` was added in v21.3. Any version of the telegram library below that would fail.
We should acknowledge that in `pyproject.toml` or somehow make these extra fields version-independent.